loki/batch/transformation.py

```diff
@@ -206,22 +258,27 @@ def apply_file(self, sourcefile, **kwargs):
         ----------
         sourcefile : :any:`Sourcefile`
             The file to transform.
+        plan_mode : bool, optional
+            If enabled, apply planning mode.
         **kwargs : optional
             Keyword arguments that are passed on to transformation methods.
         """
         if not isinstance(sourcefile, Sourcefile):
             raise TypeError('Transformation.apply_file can only be applied to Sourcefile object')
 
-        if sourcefile._incomplete:
-            raise RuntimeError('Transformation.apply_file requires Sourcefile to be complete')
-
         item = kwargs.pop('item', None)
         items = kwargs.pop('items', None)
         role = kwargs.pop('role', None)
         targets = kwargs.pop('targets', None)
 
         # Apply file-level transformations
-        self.transform_file(sourcefile, item=item, role=role, targets=targets, items=items, **kwargs)
+        if plan_mode:
+            self.plan_file(sourcefile, item=item, role=role, targets=targets, items=items, **kwargs)
+        else:
+            if not plan_mode and sourcefile._incomplete:
```

Comment:
```suggestion
            if sourcefile._incomplete:
```

double-checking for `plan_mode` to be `False`?!